casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -664,4 +664,26 @@ object BlockAPI {
     )
   }
 
+  sealed trait LatestBlockMessageError     extends Throwable
+  final case object ValidatorReadOnlyError extends LatestBlockMessageError
+  final case object NoBlockMessageError    extends LatestBlockMessageError
+
+  def getLatestMessage[F[_]: Sync: EngineCell: Log]: F[ApiErr[BlockMetadata]] = {
+    val errorMessage =
+      "Could not execute exploratory deploy, casper instance was not available yet."
+    EngineCell[F].read >>= (
+      _.withCasper(
+        implicit casper =>
+          for {
+            validatorOpt     <- casper.getValidator
+            validator        <- validatorOpt.liftTo[F](ValidatorReadOnlyError)
```

Comment:
Are we making a distinction between readonly and non-readonly validators here?

Reply:
In read-only node there is no validator identity (`casper.getValidator` returns an Option) so this function is not total. If validator is not present, the last message cannot be also and this is marked with `ValidatorReadOnlyError` exception.

Reply:
First, my understanding of read-only node is that a node can not propose a block.
As for the definition above, there is also another situation when a node with validator-identity can be read-only -> a node in the current post-state of tuplespace is not an active validator can be a read-only node. https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala#L728-L738 some related codes I wrote.
Just want to mention that.

Aside from the conversations above, we need DOCS for terminology.